WordPress/gutenberg
# Escape Editable HTML

## Description

Fixes #16252.
Fixes #13218.
Alternative to #17789.

General problem: currently we only escape "lone" ampersands (`&` => `&amp;`) in editable text. This is a bit strange, since if you create some text, you'd expect e.g. `&amp;` to be rendered as such. This is NOT the case. `&amp;` needs to be converted to `&amp;amp;` in order to be rendered as `&amp;`. The same is true for any other HTML entity.

Also removes a unnecessary layer of escaping and unescaping from the code block. The attribute source is of the type `text`, which already unescapes entities. In the code block, we only need to make sure the value from `PlainText` is properly escaped for use in normal HTML elements (since `PlainText` is a `textarea` and its value is unescaped.

## How has this been tested?

* Create a paragraph block and type `&hellip;`. Preview the post. You should see `&hellip;` and not `…`.
* Create a code block and type `&hellip;`. Save. Reload the place. The block should be valid.

## Screenshots 

## Types of changes





## Checklist:
